src/set-precision.ts

```diff
@@ -1,5 +1,11 @@
-export function setPrecision(value: number, precision: number) {
-  const precisionMultiplier = Math.pow(10, precision);
-
-  return Math.floor(value * precisionMultiplier) / precisionMultiplier;
+export function setPrecision(
+  value: number,
+  maxPrecision: number,
+  roundUp: boolean,
+) {
+  const precisionMultiplier = Math.pow(10, maxPrecision);
+  const roundedNumber = roundUp
+    ? Math.round(value * precisionMultiplier) / precisionMultiplier
```

Comment:
This is a bit odd because I may declare `roundUp: true` and still get a number that has been rounded down since `Math.round` goes to the nearest integer. What if the option was something like `roundingRule: 'up' | 'down' | 'auto' = 'down'`? And then we do `Math.ceil` / `Math.floor` / `Math.round`, respectively